src/awkward/nplike.py

```diff
@@ -406,6 +406,7 @@ def to_rectilinear(self, array, *args, **kwargs):
         if isinstance(
             array,
             (
+                numpy.ndarray,
```

Comment:
Maybe this should be in its own `if` statement? The implementation would be a pass-through, rather than going through `to_numpy`. This works, but it might be more comprehensible, from a maintenance perspective, if it's singled out as a distinct case from the Awkward classes.

Reply:
This is a good suggestion!